api/src/utils/date_utils.py

```diff
@@ -0,0 +1,16 @@
+from typing import Final
+import re
+
+iso_pattern: Final[str] = (
+    r"^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01])T([01]\d|2[0-3]):([0-5]\d):([0-5]\d)("
+    r"\.\d+)?(Z|[+-](["
+    r"01]\d|2[0-3]):([0-5]\d))?$"
+)
+
+
+def valid_iso_date(date_string: str):
```

Comment:
Nit-picking: we can mark date_string as `Optional`.
```suggestion
def valid_iso_date(date_string: Optional[str]):
```